plugin/pulpcore/plugin/changeset/iterator.py

```diff
@@ -181,34 +184,102 @@ def __iter__(self):
 
 class ArtifactIterator(Iterable):
     """
-    Iterate the content and flatten the artifacts.
-    Ensure that at least (1) artifact is yielded for each content.
+    Iterate `PendingContent` and foreach artifact, replace the DB model instance
+    with an instance fetched from the DB (when found).  Ensure that at
+    least (1) artifact is yielded for each content.
+
+    Attributes:
+        input (iterable): An iterable of PendingContent.
     """
 
     def __init__(self, content):
         """
         Args:
-            content (Iterable): An Iterable of RemoteContent.
+            content (iterable): An iterable of PendingContent.
         """
-        self.content = content
+        self.input = content
+
+    @staticmethod
+    def _batch_q(artifacts):
+        """
+        Build a query for the specified batch of artifacts.
+
+        Args:
+            artifacts (tuple): A batch of artifacts.  Each is: PendingArtifact.
+
+        Returns:
+            Q: The built query.
+        """
+        q = Q()
+        for artifact in (a for a in artifacts if a.model):
```

Comment:
How about testing that more directly with ``if not isinstance(a, NopPendingArtifact)`` ?